Updated summary charts howto, added color palette howto. (#1818)
ubccr/xdmod: docs/howto-summary-charts.md

The easiest way to create a chart to add to each user's list of charts is interactively, 
using Metric Explorer's JSON export function.

* Login to Open XDMoD with a user account that has the "Developer" role.
* Create a chart in the Metric Explorer.
* Open the "Chart Options" context menu by left clicking in the chart.
* Select the "View chart json" menu item.

To change the charts which appear in the Summary tab for all users, you need to update the
`roles.json` file (`/etc/xdmod/roles.json` if you installed the RPM or
`PREFIX/etc/roles.json` if you did a manual install).

The default summary charts (the ones included with a base intall of Open XDMoD) are included implicity 
for any user type if `summary_charts` is empty in `roles.json` for that role. As such, 
adding a chart to the `summary_charts` field for the role `pub`, which is non-logged-in users,
will override the default charts.

However, the default summary charts are also explicitly defined as added charts
for roles which inherit the `default` role (which is, by default, any logged-in users) 
by charts stored in `roles.d/jobs.json` (`/etc/xdmod/roles.d/jobs.json` if you installed the RPM or

> `PREFIX/etc/roles.d/jobs.json` if you did a manual install) - to remove these default charts from 

Reviewer: Just wanted to note that charts can be explicitly defined in multiple files in `roles.d/`. It might be more clear to hint that these other files can be customized (such as `roles.d/cloud.json`) not just `roles.d/jobs.json`.

Author: After looking through the configuration files more, I've significantly rewrote this section of the article. I originally wrote it through the lens of "dang, just let me set my defaults!" rather than actually looking at where anything should/could go.

title: HOWTO Change Summary Page Charts
---

If a user wants to add a chart to their personal Summary tab, they can use Metric Explorer to create a chart,
then open the "Chart Options" context menu by left clicking in the chart, then check the "Show in Summary
tab" checkbox.

To change the charts that appear in users' Summary tabs by default, you'll need to create
a chart and add its JSON definition to the XDMoD configuration files.

The easiest way to create a JSON definition for a chart is interactively, 
using Metric Explorer's JSON export function:

* Login to Open XDMoD with a user account that has the "Developer" role.
* Create a chart in the Metric Explorer.
* Open the "Chart Options" context menu by left clicking in the chart.
* Select the "View chart json" menu item.

Default summary charts are defined as lists of charts within the configuration files for user roles.
These are located in two different sets of locations: in the top level `roles.json` file 
(`/etc/xdmod/roles.json` if you installed the RPM or `PREFIX/etc/roles.json` if you did a manual install), 

> and in any `.json` file located in `roles.d` (`/etc/xdmod/roles.d/*.json` or `PREFIX/xdmod/roles.d/*.json` respectively).

Reviewer: ```suggestion
and in some `.json` files located in `roles.d` (`/etc/xdmod/roles.d/` or `PREFIX/xdmod/roles.d/` respectively). The files that contain `+summary_charts` are typically organized into different realm configuration files such as `roles.d/jobs.json` and `roles.d/cloud.json`.
```
I just wanted to specify this because of `dashboard.json` and `statistics_format.json` in `roles.d/`.

The base summary charts (the ones included with a install of Open XDMoD) are included implicity 
for any user role if the field `summary_charts` is absent for that role in both `roles.json` and

> any files in `roles.d`. If `summary_charts` contains any charts, those override displaying any of the summary charts.

Reviewer: ```suggestion
in `roles.d`. If `summary_charts` contains any charts, those override displaying any of the summary charts.
```

title: HOWTO Change Summary Page Charts
---

If a user wants to add a chart to their personal Summary tab, they can use Metric Explorer to create a chart,
then open the "Chart Options" context menu by left clicking in the chart, then check the "Show in Summary
tab" checkbox.

To change the charts that appear in users' Summary tabs by default, you'll need to create
a chart and add its JSON definition to the XDMoD configuration files.

The easiest way to create a JSON definition for a chart is interactively,
using Metric Explorer's JSON export function:

* Login to Open XDMoD with a user account that has the "Developer" role.
* Create a chart in the Metric Explorer.
* Open the "Chart Options" context menu by left clicking in the chart.
* Select the "View chart json" menu item.

Default summary charts are defined as lists of charts within the configuration files for user roles.
These are located in two different sets of locations: in the top level `roles.json` file
(`/etc/xdmod/roles.json` if you installed the RPM or `PREFIX/etc/roles.json` if you did a manual install),

> and in some `.json` files located in `roles.d` (`/etc/xdmod/roles.d/` or `PREFIX/xdmod/roles.d/` respectively). The files that contain `+summary_charts` are typically organized into different realm configuration files such as `roles.d/jobs.json` and `roles.d/cloud.json`.

Author: Should I switch it back to the language describing _any_ `.json` file in that folder being parsed for roles?

The base summary charts (the ones included with a install of Open XDMoD) are included implicity
for any user role if the field `summary_charts` is absent for that role in both `roles.json` and
in `roles.d`. If `summary_charts` contains any charts, those override displaying any of the summary charts.

By default:

* The role "pub" - which is applied to any users who are not logged in - has no `summary_charts` field and so implictly uses the base summary charts.
* The role "default" - which all other user roles inherit - has an empty `summary_charts` field in `roles.json`, but

>   has explicitly defined summary charts appended in `roles.d/jobs.json` equivalent to the base summary charts.

Reviewer: ```suggestion
  has explicitly defined summary charts appended in `roles.d/jobs.json` or your custom summary chart `.json` file that are equivalent to the base summary charts.
```

]
```

> The export functionality from Metric Explorer does not, by default,

Reviewer: We are aware of the issues surrounding `global_filters` within the Summary tab. I have a minor suggestion that it may be more concise to mention that an empty `global_filters` field is required in the chart JSON due to a known issue

Author: got it - changed the wording and I think it's clearer now, though I'm unsure if it's more concise.
